### DISCUSSION TITLE
How to handle same user registration from different OAuth

### DISCUSSION THREAD

**Discussion by jd-solanki** [fastapi-users/fastapi-users#549] [Ideas]
Hi @frankie567 , I recently worked with Django and I am now moving to FastAPI and for user management I discovered your package. It is really helpful and time-saving :heart: 

I checked docs and I am unable to figure out does fastapi-users has the ability to handle the same user registration from multiple OAuth services.

E.g.
App provides Google and GitHub for authentication.

Let's assume john uses the same email for google and Github.

One day 1 john logins using Google Oauth and get registered in our app and next month he forgets which Oauth he used and this time he tries GitHub. In this case, how many users we will have?

I guess, fastapi-users will create two users under the same email, right?
But as per the user perspective, I should be get logged in to my old account and my user data will have two OAuth service registred. So, we don't have duplicate users in the Database.

What do you say? In Django i created two tables for the user,
1. User (Which just holds user common data like primary email, id/uuid, is_superuser, etc.)
2. UserProfile (This will create a new row for each OAuth service - It will hold OAuth service email, avatar, username, is_connected, etc.)

Regards.

**Comment by frankie567** [comment]
Hi there!

FastAPI Users enforces emails _uniqueness_ in the database. Thus, if someone is already registered with that e-mail and try to authenticate with an OAuth service we didn't know, it'll be linked to this user and authenticated. A single user can have multiple OAuth accounts in the database.

You can check the relevant logic here:

https://github.com/frankie567/fastapi-users/blob/e7ceb1569c2f72a490b72574c12cd74acd7f7a86/fastapi_users/router/oauth.py#L117-L133

**Reply by jd-solanki** [reply]
Thanks so much @frankie567 :pray: 

I will try it for sure :+1:
